### ISSUE TITLE
[Request] Manually add/run already installed app

### ISSUE THREAD

**Issue by hthre7** [bottlesdevs/Bottles#394]
**Is your feature request related to a problem? Please describe.**
as lutris manual adding feature just detect game directory

**Comment by mirkobrombin** [comment]
I did not understand

**Comment by hthre7** [comment]
so copy game file to bottles directory manuall and bottle should detect the .exe or add location of .exe manually

**Comment by mirkobrombin** [comment]
Bottles use lnk files from the Start Menu to find installed programs. https://docs.usebottles.com/bottles/programs
Then copy the .lnk file as well and place it in the correct Start Menu folder and Bottles will detect it

**Comment by drws** [comment]
I see two issues here. The first being with importing an existing Wine prefix into Bottles, which can also be done manually as mentioned. The README file lists import compatibility with POL, Lutris and Bottles v1, but not an ordinary Wine prefix. Probably it works and only needs to be listed in the README?

The second one you just mentioned and it seems more dire. There are Windows installers that do not provide .lnk files, so Start Menu folder detection is not 100% reliable.

**Comment by mirkobrombin** [comment]
> I see two issues here. The first being with importing an existing Wine prefix into Bottles, which can possibly be hacked manually. The [documentation](https://docs.usebottles.com/bottles/import-from-other-managers) and README list import compatibility with POL, Lutris and Bottles v1, but not with an ordinary Wine prefix. Probably this is a feature request to implement it.

Bottles detects the prefixes from Lutris, POL and .wine (in homedir) directories. In the future it will allow you to choose custom routes from which to import prefixes.

> The second one you just mentioned and it seems more dire. There are Windows installers that do not provide .lnk files, so Start Menu folder detection is not 100% reliable.

If a program does not appear in the Windows Start menu, it is because it does not have to be listed to the user. So he expects the same behavior on Bottles.

**Comment by drws** [comment]
> If a program does not appear in the Windows Start menu, it is because it does not have to be listed to the user. So he expects the same behavior on Bottles.

Not necessarily. Consider an older piece of software or one that is (or stayed) in development, which simply does not provide a .lnk. Also, I've seen userspace programs back in the day that only installed an icon in the Desktop folder, but not in Start Menu one. Or the ones that launch through CD autorun. Oh, Winblows.

Although it seems that in the recent years Windows software installers became a bit more orderly so .lnk in Start Menu _kinda became standard_. But Wine users may also be running older software with dumb and/or custom installers. And if you aim for best compatibility possible...

**Comment by mirkobrombin** [comment]
> > If a program does not appear in the Windows Start menu, it is because it does not have to be listed to the user. So he expects the same behavior on Bottles.
> 
> Not necessarily. Consider an older piece of software or one that is (or stayed) in development, which simply does not provide a .lnk. Also, I've seen userspace programs back in the day that only installed an icon in the Desktop folder, but not in Start Menu one. Or the ones that launch through CD autorun. Oh, Winblows.
> 
> Although it seems that in the recent years Windows software installers became a bit more orderly so .lnk in Start Menu _kinda became standard_. But Wine users may also be running older software with dumb and/or custom installers. And if you aim for best compatibility possible...

I can allow the user to pick an executable and add to the program list but it requires some work 🤔

**Comment by drws** [comment]
Yes, but it would provide a robust way to fix such software :) Thank you for the consideration.

**Comment by hthre7** [comment]
so after trying the new version still cant detect app.exe path (I added the executable manually) but bottle should detect the same directory as app directory to get necessary library/dlls of that app to be able to launch(when try to browse path I get this error `unable to detect URI-scheme of "app.exe"`) .
also already installed isn't a portable-app it as it self-explanatory just installed (for example if you an app already installed in your pc you could copy it entire directory from(program files) to different pc it should work assuming it have the basic system library(such .net framework ,etc )

**Comment by mirkobrombin** [comment]
> so after trying the new version still cant detect app.exe path (I added the executable manually and its there in program tab in bottles) but bottle should detect the same directory as app directory to get necessary library/dlls of that app to be able to launch(when try to browse path I get this error `unable to detect URI-scheme of "app.exe"`) .
> also already installed isn't a portable-app it as it self-explanatory just installed (for example if you an app already installed in your pc you could copy it entire directory from(program files) to different pc it should work assuming it have the basic system library(such .net framework ,etc )

What you are asking for cannot be integrated into Bottles. Lutris was born as an application manager (regardless of the runner it uses to run it), so you can add any executable, also setting the Path to the prefix to use as the environment for all libraries needed by the executable. 

Bottles is strictly a prefix manager (its own prefixes). If you want to use executables from other prefixes, you have to import that prefix into Bottles but even that doesn't necessarily work without a bottle.yml containing all the instruction that Bottles needs to run executables in the prefix.

The action of adding custom executables to the program list, should only be used when Bottles does not automatically detect the program, or if you want to add standalone executables. This was originally supposed to be limited to the Path of the bottle only but sadly it is not possible to handle this thing in GTK.

In short: I am not refusing to integrate this thing but it is not possible for how Bottles was designed and developed.

**Comment by drws** [comment]
I'm not sure @hthre7 wants to run an executable from another prefix. Maybe what's missing is changing the directory to the folder in which exe is located before running it, if that's not already done (similar to #397).

**Comment by mirkobrombin** [comment]
> I'm not sure @hthre7 wants to run an executable from another prefix. Maybe what's missing is changing the directory to the folder in which exe is located before running it, if that's not already done (similar to #397).

I may have misunderstood. But this does not completely solve the problem because that way you only change the cwd but the prefix will still continue to search for libraries in the bottle. Also, even if you don't change the cwd, the executable should still automatically look for its libraries in the directory from where it was run. If this is not the case then the problem should be solved with #397.

**Comment by hthre7** [comment]
> I'm not sure @hthre7 wants to run an executable from another prefix. Maybe what's missing is changing the directory to the folder in which exe is located before running it, if that's not already done (similar to #397).
actually what I did just copied already installed to bottle prefix so I think .yml file 'mirkobrombin' is needed
